blog/src/__tests__/BlogDetails.test.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { screen } from '@testing-library/react';
+import { MemoryRouter } from 'react-router-dom';
+import { renderWithProviders } from '../utils/wrappertesting';
+import BlogDetails from '../components/BlogDetails/BlogDetails';
+import { greater } from '../utils/helperfunctions';
+
+describe('Unit tests for BlogDetails Components', () => {
+  it('Snapshot test for BlogDetails component', () => {
+    const { asFragment } = renderWithProviders(<MemoryRouter><BlogDetails /></MemoryRouter>);
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('Dom test for rendering BlogDetails', () => {
+    renderWithProviders(<MemoryRouter><BlogDetails /></MemoryRouter>);
+    const detailseElement = screen.getByTestId('details');
+    expect(detailseElement).toBeInTheDocument();
+  });
+
+  it('Image renders in BlogDetails component', () => {
+    renderWithProviders(<MemoryRouter><BlogDetails blogDetails={greater[0]} key={greater.id} /></MemoryRouter>);
+    const image = screen.getByTestId('Image');
+    expect(image).toBeInTheDocument();
+  });
+
+  it('Image have ALT', () => {
+    renderWithProviders(<MemoryRouter><BlogDetails blogDetails={greater[0]} key={greater.id} /></MemoryRouter>);
+    const img = screen.getByTestId('Image');
+    expect(img).toHaveAttribute('alt', 'Article');
+  });
+
+  it('Correct title is rendered as Heading', () => {
+    renderWithProviders(<MemoryRouter><BlogDetails blogDetails={greater[0]} key={greater.id} /></MemoryRouter>);
+    const titleheading = screen.getByTestId('titleHeading');
+    expect(titleheading).toBe('10 Tips for Budget-Friendly Travel');
+  });
+
+  it('Discription is rendered', () => {
+    renderWithProviders(<MemoryRouter><BlogDetails blogDetails={greater[0]} key={greater.id} /></MemoryRouter>);
+    const discription = screen.getByTestId('titleHeading');
+    expect(discription.textContent.length).toBeGreaterThan(0);
```

Comment:
I recommend checking the description equals description of `greater[0]` blog.
For Example:
`expect(titleheading).toBe(greater[0].content);`